app/src/main/java/com/epfl/drawyourpath/userProfileCreation/UserNameTestAndSetFragment.kt

```diff
@@ -0,0 +1,112 @@
+
+        testUserNameButton.setOnClickListener {
+            usernameAvaibility(database, inputUserName.text.toString(), showTestResult)
+        }
+
+        val setUserNameButton: Button =
+            view.findViewById(R.id.setUserName_button_userProfileCreation)
+        setUserNameButton.setOnClickListener {
+            //try to set the userName to the database
+            val usernameStr = inputUserName.text.toString()
+            val testUsername = usernameAvaibility(database, usernameStr, showTestResult)
+            val previousActivity = activity
+            if (previousActivity != null && testUsername) {
+                database.setUserName(usernameStr)
+                val fragManagement = previousActivity.supportFragmentManager.beginTransaction()
+                val dataToPersoInfoFrag: Bundle = Bundle()
+                //data to transmit to the PersonalInfoFragment(username + isTest)
+                dataToPersoInfoFrag.putBoolean("isRunningTestForDataBase", isTest)
+                dataToPersoInfoFrag.putString("userName", inputUserName.text.toString())
+                val persoInfoFrag = PersonalInfoFragment()
+                persoInfoFrag.arguments = dataToPersoInfoFrag
+                fragManagement.replace(R.id.userName_frame, persoInfoFrag).commit()
+            }
+        }
+    }
+}
+
+/**
+ * Helper function that display a message with outputMessage to the user on the UI and return a boolean that indicate
+ * if the username is available
+ * @param username username with the availability to tes
+ * @param database used to test if the username is available
+ * @param outputMessage editText used to show an error message to the user on the UI if the username is not available
+ * @return true if the username is available, and false otherwise
+ */
+private fun usernameAvaibility(database: Database, username: String, outputMessage: TextView): Boolean{
+    if (username == "") {
+        outputMessage.text = buildString { append("The username can't be empty !") }
+        outputMessage.setTextColor(Color.RED)
+        return false
+    }
+
+    val unAvailableOutput: String = buildString {
+        append("*The username ")
+        append(username)
+        append(" is NOT available !")
+    }
+    val availableOutput: String = buildString {
+        append("*The username ")
+        append(username)
+        append(" is available !")
+    }
+    val future = database.isUserNameAvailable(username)
+
+    //since the orTimeout require an API level 33(we are in min API level 28), we can't use it
+    val durationFuture = future.thenAccept {
+        outputMessage.text = buildString {
+            append("*The username ")
+            append(username)
+            append(" is ")
+            append(if (!it) "NOT " else "")
+            append("available !")
+        }
+        outputMessage.setTextColor(if (!it) Color.GREEN else Color.RED)
```

Comment:
It should be 
```kotlin
outputMessage.setTextColor(if (it) Color.GREEN else Color.RED)
```
to have a green text when it is available.

Reply:
I also correct it, thanks. I see the error too when I test in the future was correctly implemented